Sources/PackageCollections/Providers/JSONPackageCollectionProvider.swift

```diff
@@ -270,22 +287,33 @@ struct JSONPackageCollectionProvider: PackageCollectionProvider {
         }
 
         public init(maximumSizeInBytes: Int64? = nil,
+                    trustedRootCertsDir: URL? = nil,
+                    sourceCertPolicies: [String: CertificatePolicyKey]? = nil,
                     maximumPackageCount: Int? = nil,
                     maximumMajorVersionCount: Int? = nil,
                     maximumMinorVersionCount: Int? = nil) {
             // TODO: where should we read defaults from?
             self.maximumSizeInBytes = maximumSizeInBytes ?? 5_000_000 // 5MB
+            self.trustedRootCertsDir = trustedRootCertsDir
+            self.sourceCertPolicies = sourceCertPolicies ?? [:]
             self.validator = JSONModel.Validator.Configuration(
                 maximumPackageCount: maximumPackageCount,
                 maximumMajorVersionCount: maximumMajorVersionCount,
                 maximumMinorVersionCount: maximumMinorVersionCount
             )
         }
+
+        func certificatePolicyKey(for source: Model.CollectionSource) -> CertificatePolicyKey? {
+            // Certificate policy is associated to a collection host
+            guard let host = source.url.host else { return nil }
+            return self.sourceCertPolicies[host]
```

Comment:
nit: you could also use functional style, i.e. `source.url.host.map { self.sourceCertPolicies[$0] }`